torch/testing/_internal/common_fsdp.py

```diff
@@ -23,30 +33,97 @@
 
 
 class FSDPInitMode(Enum):
-    # Move model to CUDA before wrap
-    CUDA_BEFORE = 1
-    # Move model to CUDA after wrap
-    CUDA_AFTER = 2
-    # Don't move model to CUDA at all.
-    CUDA_NEVER = 3
-
-
-def _validate(model, process_group, assert_fn):
-    module_states = [param.detach().cpu() for param in model.parameters()]
-    module_states.extend([buffer.detach().cpu() for buffer in model.buffers()])
+    # No FSDP wrapping
+    NO_FSDP = auto()
+    # FSDP recursive wrapping
+    RECURSIVE = auto()
+    # TODO: FSDP non-recursive wrapping
+    # NONRECURSIVE = auto()
```

Comment:
The idea is that when I am testing the non-recursive wrapping, I implement a branch for `fsdp_init_mode == FSDPInitMode.NONRECURSIVE` for each model in `common_fsdp.py` and then pass `fsdp_init_mode=FSDPInitMode.NONRECURSIVE` from each existing test. There should not need to be any other changes.